airflow/api_connexion/endpoints/connection_endpoint.py

```diff
@@ -129,3 +133,21 @@ def post_connection(session):
         session.commit()
         return connection_schema.dump(connection)
     raise AlreadyExists(detail=f"Connection already exist. ID: {conn_id}")
+
+
+@security.requires_access([(permissions.ACTION_CAN_CREATE, permissions.RESOURCE_CONNECTION)])
+def test_connection():
+    """Test a connection"""
+    body = request.json
+    try:
+        data = connection_schema.load(body)
+        dummy_conn_id = get_random_string()
+        data['conn_id'] = dummy_conn_id
+        conn = Connection(**data)
+        conn_env_var = f'AIRFLOW_CONN_{dummy_conn_id.upper()}'
+        os.environ[conn_env_var] = conn.get_uri()
+        status, message = conn.test_connection()
+        del os.environ[conn_env_var]
```

Comment:
In case of an exception this env var is not deleted -- we should delete this in a `finally` block.